app/src/main/java/com/magdamiu/androidfundamentalsmai2021/MainActivity.java

```diff
@@ -1,131 +1,55 @@
-    private void setAndroidVersions() {
-        androidVersions = new ArrayList<>();
-        androidVersions.add("cupcake");
-        androidVersions.add("eclair");
-        androidVersions.add("pie");
-        androidVersions.add("donut");
-        androidVersions.add("kitkat");
-    }
-
-    // step 2: initialise adapter for our spinner
-    // ArrayAdapter is a default adapter
-    private void initialiseSpinnerAdapter() {
-        spinnerAdapter = new ArrayAdapter<>(MainActivity.this,
-                android.R.layout.simple_spinner_item,
-                androidVersions);
-    }
-
-    // step 3: set the adapter to our spinner
-    private void setSpinnerAdapter() {
-        spinnerAndroidVersions = findViewById(R.id.spinnerAndroidVersions);
-        spinnerAndroidVersions.setAdapter(spinnerAdapter);
-    }
-
-    private void handlingSpinnerListener() {
-        spinnerAndroidVersions.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
-            @Override
-            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-                String selectedAndroidVersion = androidVersions.get(i);
-                Toast.makeText(MainActivity.this, selectedAndroidVersion, Toast.LENGTH_LONG).show();
-            }
-
-            @Override
-            public void onNothingSelected(AdapterView<?> adapterView) {
-
-            }
-        });
-    }
-
-    // webview load url
-    private void loadUrl() {
-        webView = findViewById(R.id.webViewSample);
-        webView.getSettings().setJavaScriptEnabled(true);
-        webView.loadUrl(ANDROID_URL);
-    }
-
-    // init views for implementing a first simple form
     private void initViews() {
-        editTextName = findViewById(R.id.editTextName);
-        buttonDisplayGreetings = findViewById(R.id.buttonDisplayGreetings);
-        textViewGreetings = findViewById(R.id.textViewGreetings);
-    }
-
-    private void displayLogs() {
-        Log.e("tag-error", "my first error log");
-        Log.v("tag-verbose", "my first verbose log");
-        Log.w("tag-warning", "my first warning log");
-    }
-
-    private void clickOnDisplayGreetingsButton() {
-        buttonDisplayGreetings.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-
+        emailLogin = findViewById(R.id.emailLogin);
+        phoneLogin = findViewById(R.id.phoneLogin);
+        checkBoxLogin = findViewById(R.id.checkboxLogin);
+        buttonLogin = findViewById(R.id.buttonLogin);
+        textLogin = findViewById(R.id.textLogin);
+    }
+
+    public void displaySubmitOnClick(View view) {
+        textLogin.setText("");
+        String email = emailLogin.getText().toString();
+        String phone = phoneLogin.getText().toString();
+        final String successSubmit = "Successfully submitted";
+        if (email.length() > 0 && Patterns.EMAIL_ADDRESS.matcher(email).find()) {
```

Comment:
To make the code more readable I suggest implementing a function to validate if a string is a valid email address. Same idea for phone number.

Reply:
solved